src/NHibernate/QueryExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections;
+using System.Collections.Generic;
+using NHibernate.Impl;
+
+namespace NHibernate
+{
+	// 6.0 TODO: consider merging into IQuery.
+	public static class QueryExtensions
+	{
+		/// <summary>
+		/// Bind the keys values of the given dictionary to named parameters of the query,
+		/// matching keys with parameter names and mapping value types to
+		/// NHibernate types using heuristics.
+		/// </summary>
+		/// <param name="query">The query on which to set the parameters.</param>
+		/// <param name="map">A dictionary of parameters values by names.</param>
+		public static IQuery SetParameters(this IQuery query, IDictionary<string, object> map)
```

Comment:
I had to change the name, since non extension methods takes precedence. Moreover the name `SetProperties` does not make much sense in my view, this one is more semantically accurate I think.

So I have added a commit to obsolete `SetProperties` in favor of a `SetParameters` overload.